Documentation/boot-flow.md

```diff
@@ -0,0 +1,71 @@
+# Boot Flow
+
+## Storage
+
+Firmware resides in a storage part. One commonly used kind for firmware is
+[flash storage](https://www.netapp.com/data-storage/what-is-flash-storage),
+often SPI NOR, but other parts are possible as well, depending on the platform.
+Those may be SPI NAND, which can hold larger amounts of data, eMMC/MMC, SD card,
+USB drives, and in split scenarios, a fraction may even reside on the network.
+
+For simplicity, we refer to a typical 16MB NOR flash in the following example.
+
+### DTFS (Device Tree File System)
+
+oreboot leaves a data structure in the storage that describes the overall layout
+in the IEEE 1275 [Device Tree](https://www.devicetree.org/) format. This allows
+for splitting up an image again and is defined per mainboard in a file named
+`fixed-dtfs.dts`.
+
+## Stages
+
+Due to the nature of platforms, firmware needs to start small. From stage to
+stage, we set up an environment such that an operating system may be booted or
+any other payload executed.
+With oreboot, we focus on embedding [LinuxBoot](https://linuxboot.org).
+
+![oreboot flash and flow](img/firmware-matryoshka.png)
+
+### Stage 1: SRAM/XIP
```

Comment:
The only issue is that on some systems the bootblock has to be distinct from this first stage, since not even the processor mode is compatible. I would still argue that we need a Stage 0: here, which can be optional?